qa/integration/services/logstash_service.rb

```diff
@@ -165,6 +165,7 @@ def wait_for_logstash
       end
       tries -= 1
     end
+    raise RuntimeError.new("failed to start Logstash after #{RETRY_ATTEMPTS} seconds. aborting..")
```

Comment:
You need to check if you ran out of retries, because right now all the execution will go through this path event if it succeeded.